### Preparing the Data
- Importing needed libraries/tools
- Reading in the Pickle Files from 'CleanData' folder
- Splitting all_players to exclude players with 'no team'

*First Portion of this will be with the entire collection of data*

In [170]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix



In [100]:
all_players = pd.read_pickle('/Users/powellshayne/Desktop/githubrepos/VictorVis/CleanData/all_players_df.pkl')
all_players

player_name         real_name      team  age    rating       dpr  \
0         ZywOo   Mathieu Herbaut  vitality   23  1.000000  0.333333   
1         Spinx      Lotan Giladi  vitality   23  0.627451  0.476190   
2         mezii  William Merriman  vitality   23  0.588235  0.428571   
3        flameZ    Shahar Shushan  vitality   23  0.607843  0.571429   
4          apEX   Dan Madesclaire  vitality   23  0.372549  0.714286   
..          ...               ...       ...  ...       ...       ...   
963   seang@res        Sean Gares   no team   23  0.137255  0.666667   
964         def   Dima Lemeshchuk   no team   23  0.098039  0.714286   
965      advent        Zhuo Liang   no team   23  0.078431  0.476190   
966       gob b       Fatih Dayik   no team   23  0.078431  0.619048   
967     netrick   Jonas Wenzelsen   no team   23  0.019608  0.809524   

          kpr      kast    impact       adr  ...  sniping_kills_percentage  \
0    0.966667  0.924370  1.000000  0.855263  ...                  0.425633   
1    0.600000  0.663866  0.529412  0.549342  ...                  0.011076   
2    0.500000  0.747899  0.441176  0.536184  ...                  0.028481   
3    0.500000  0.689076  0.602941  0.549342  ...                  0.014241   
4    0.466667  0.352941  0.455882  0.523026  ...                  0.020570   
..        ...       ...       ...       ...  ...                       ...   
963  0.133333  0.302521  0.176471  0.200658  ...                  0.052215   
964  0.133333  0.386555  0.147059  0.200658  ...                  0.012658   
965  0.000000  0.537815  0.000000  0.000000  ...                  0.007911   
966  0.033333  0.319328  0.132353  0.171053  ...                  0.037975   
967  0.033333  0.159664  0.147059  0.180921  ...                  0.011076   

     sniping_rounds_with_kills_percentage  sniping_multi_kill_rounds  \
0                                0.505155                   0.444444   
1                                0.013746                   0.008889   
2                                0.030928                   0.022222   
3                                0.017182                   0.008889   
4                                0.020619                   0.011111   
..                                    ...                        ...   
963                              0.089347                   0.088889   
964                              0.010309                   0.010000   
965                              0.006873                   0.003333   
966                              0.024055                   0.033333   
967                              0.010309                   0.008889   

     sniping_opening_kills_per_round  utility_score  utility_damage_per_round  \
0                            0.37500       0.609195                  0.180239   
1                            0.00125       0.252874                  0.173724   
2                            0.01250       0.793103                  0.277959   
3                            0.00250       0.839080                  0.302932   
4                            0.02500       0.551724                  0.232356   
..                               ...            ...                       ...   
963                          0.07500       0.333333                  0.292074   
964                          0.00000       0.563218                  0.146580   
965                          0.00000       0.735632                  0.205212   
966                          0.01250       0.747126                  0.280130   
967                          0.00875       0.873563                  0.422367   

     utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
0                        0.474026                          0.652666   
1                        0.428571                          0.410339   
2                        0.422078                          0.781906   
3                        0.500000                          0.765751   
4        

In [101]:
team_players = all_players[all_players['team'] != 'no team']
team_players

player_name           real_name            team  age    rating       dpr  \
0         ZywOo     Mathieu Herbaut        vitality   23  1.000000  0.333333   
1         Spinx        Lotan Giladi        vitality   23  0.627451  0.476190   
2         mezii    William Merriman        vitality   23  0.588235  0.428571   
3        flameZ      Shahar Shushan        vitality   23  0.607843  0.571429   
4          apEX     Dan Madesclaire        vitality   23  0.372549  0.714286   
..          ...                 ...             ...  ...       ...       ...   
580       dobbo      William Dobson             exo   23  0.313725  0.523810   
581      S0tF1k  Dmitry Forostyanko  spirit academy   23  0.196078  0.666667   
582      coolio      András Fercsák     navi junior   23  0.254902  0.476190   
583      Psycho     Alexandr Zlobin            dash   23  0.274510  0.714286   
584        bobz         Pedro Rocha    mibr academy   23  0.176471  0.761905   

          kpr      kast    impact       adr  ...  sniping_kills_percentage  \
0    0.966667  0.924370  1.000000  0.855263  ...                  0.425633   
1    0.600000  0.663866  0.529412  0.549342  ...                  0.011076   
2    0.500000  0.747899  0.441176  0.536184  ...                  0.028481   
3    0.500000  0.689076  0.602941  0.549342  ...                  0.014241   
4    0.466667  0.352941  0.455882  0.523026  ...                  0.020570   
..        ...       ...       ...       ...  ...                       ...   
580  0.200000  0.462185  0.161765  0.184211  ...                  0.018987   
581  0.233333  0.344538  0.294118  0.312500  ...                  0.045886   
582  0.100000  0.470588  0.058824  0.101974  ...                  0.004747   
583  0.200000  0.361345  0.191176  0.282895  ...                  0.009494   
584  0.033333  0.268908  0.132353  0.151316  ...                  0.009494   

     sniping_rounds_with_kills_percentage  sniping_multi_kill_rounds  \
0                                0.505155                   0.444444   
1                                0.013746                   0.008889   
2                                0.030928                   0.022222   
3                                0.017182                   0.008889   
4                                0.020619                   0.011111   
..                                    ...                        ...   
580                              0.017182                   0.022222   
581                              0.041237                   0.033333   
582                              0.003436                   0.005556   
583                              0.006873                   0.008889   
584                              0.006873                   0.007778   

     sniping_opening_kills_per_round  utility_score  utility_damage_per_round  \
0                            0.37500       0.609195                  0.180239   
1                            0.00125       0.252874                  0.173724   
2                            0.01250       0.793103                  0.277959   
3                            0.00250       0.839080                  0.302932   
4                            0.02500       0.551724                  0.232356   
..                               ...            ...                       ...   
580                          0.00750       0.551724                  0.147666   
581                          0.01250       0.827586                  0.261672   
582                          0.00250       0.643678                  0.187839   
583                          0.00000       0.931034                  0.350706   
584                          0.00250       0.804598                  0.199783   

     utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
0                        0.474026                          0.652666   
1                        0.428571                          0.410339   
2                        0.422078                      

In [102]:
team_players = team_players.set_index('player_name')
team_players

real_name            team  age    rating       dpr  \
player_name                                                                
ZywOo           Mathieu Herbaut        vitality   23  1.000000  0.333333   
Spinx              Lotan Giladi        vitality   23  0.627451  0.476190   
mezii          William Merriman        vitality   23  0.588235  0.428571   
flameZ           Shahar Shushan        vitality   23  0.607843  0.571429   
apEX            Dan Madesclaire        vitality   23  0.372549  0.714286   
...                         ...             ...  ...       ...       ...   
dobbo            William Dobson             exo   23  0.313725  0.523810   
S0tF1k       Dmitry Forostyanko  spirit academy   23  0.196078  0.666667   
coolio           András Fercsák     navi junior   23  0.254902  0.476190   
Psycho          Alexandr Zlobin            dash   23  0.274510  0.714286   
bobz                Pedro Rocha    mibr academy   23  0.176471  0.761905   

                  kpr      kast    impact       adr  rating_is_missing  ...  \
player_name                                                             ...   
ZywOo        0.966667  0.924370  1.000000  0.855263                  0  ...   
Spinx        0.600000  0.663866  0.529412  0.549342                  0  ...   
mezii        0.500000  0.747899  0.441176  0.536184                  0  ...   
flameZ       0.500000  0.689076  0.602941  0.549342                  0  ...   
apEX         0.466667  0.352941  0.455882  0.523026                  0  ...   
...               ...       ...       ...       ...                ...  ...   
dobbo        0.200000  0.462185  0.161765  0.184211                  0  ...   
S0tF1k       0.233333  0.344538  0.294118  0.312500                  0  ...   
coolio       0.100000  0.470588  0.058824  0.101974                  0  ...   
Psycho       0.200000  0.361345  0.191176  0.282895                  0  ...   
bobz         0.033333  0.268908  0.132353  0.151316                  0  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
ZywOo                        0.425633                              0.505155   
Spinx                        0.011076                              0.013746   
mezii                        0.028481                              0.030928   
flameZ                       0.014241                              0.017182   
apEX                         0.020570                              0.020619   
...                               ...                                   ...   
dobbo                        0.018987                              0.017182   
S0tF1k                       0.045886                              0.041237   
coolio                       0.004747                              0.003436   
Psycho                       0.009494                              0.006873   
bobz                         0.009494                              0.006873   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
ZywOo                         0.444444                          0.37500   
Spinx                         0.008889                          0.00125   
mezii                         0.022222                          0.01250   
flameZ                        0.008889                          0.00250   
apEX                          0.011111                          0.02500   
...                                ...                              ...   
dobbo                         0.022222                          0.00750   
S0tF1k                        0.033333                          0.01250   
coolio                        0.005556                          0.00250   
Psycho                        0.008889                          0.00000   
bobz                          0.007778                          0.00250   

 

In [103]:
team_players.columns
team_players = team_players.drop(columns=['real_name', 'team', 'rating_is_missing'])
team_players.columns

Index(['age', 'rating', 'dpr', 'kpr', 'kast', 'impact', 'adr', 'total_kills',
       'total_deaths', 'rounds_played', 'kd_ratio', 'maps_played',
       'kills_per_round', 'deaths_per_round', 'assists_per_round',
       'saved_by_teammate_per_round', 'saved_teammates_per_round',
       'damage_per_round', 'grenade_damage_per_round', 'headshot_percentage',
       'firepower_score', 'firepower_kills_per_round',
       'firepower_damage_per_round', 'firepower_kills_per_round_win',
       'firepower_damage_per_round_win', 'firepower_rounds_with_kill',
       'firepower_rounds_with_multi_kill', 'opening_score',
       'opening_kills_per_round', 'opening_deaths_per_round',
       'opening_attempts', 'opening_success',
       'win_percentage_after_opening_kill', 'opening_attacks_per_round',
       'clutching_score', 'clutch_points_per_round',
       'clutch_1on1_win_percentage', 'clutch_time_alive_per_round',
       'clutch_saves_per_round_loss', 'clutch_last_alive_percentage',
       'entryin

## Creating a Binary Column
- This will help with train_test_split
- '1's and '0's for rating

In [104]:
average_rating = team_players['rating'].mean()
average_rating

0.4816490698843638

In [105]:
team_players['rating'] = (team_players['rating'] > average_rating).astype(int)

In [106]:
team_players['rating'].value_counts()

rating
1    309
0    276
Name: count, dtype: int64

In [107]:
display(team_players.head(15))
display(team_players.tail(15))

age  rating       dpr       kpr      kast    impact       adr  \
player_name                                                                  
ZywOo         23       1  0.333333  0.966667  0.924370  1.000000  0.855263   
Spinx         23       1  0.476190  0.600000  0.663866  0.529412  0.549342   
mezii         23       1  0.428571  0.500000  0.747899  0.441176  0.536184   
flameZ        23       1  0.571429  0.500000  0.689076  0.602941  0.549342   
apEX          23       0  0.714286  0.466667  0.352941  0.455882  0.523026   
s1mple        23       1  0.428571  1.000000  0.798319  0.867647  0.792763   
w0nderful     23       1  0.190476  0.733333  0.899160  0.573529  0.513158   
jL            23       1  0.619048  0.600000  0.579832  0.573529  0.575658   
b1t           23       1  0.428571  0.500000  0.672269  0.529412  0.467105   
iM            23       1  0.714286  0.600000  0.470588  0.617647  0.667763   
Aleksib       23       0  0.523810  0.233333  0.563025  0.338235  0.368421   
B1ad3         23       0  0.904762  0.100000  0.000000  0.132353  0.154605   
sh1ro         23       1  0.000000  0.733333  1.000000  0.661765  0.559211   
donk          23       1  0.619048  0.966667  0.815126  0.926471  0.888158   
zont1x        23       1  0.523810  0.433333  0.680672  0.500000  0.532895   

             total_kills  total_deaths  rounds_played  ...  \
player_name                                            ...   
ZywOo           0.535451      0.360651       0.419567  ...   
Spinx           0.181227      0.141909       0.152015  ...   
mezii           0.208649      0.180832       0.196674  ...   
flameZ          0.156083      0.141453       0.138413  ...   
apEX            0.838105      0.891845       0.873289  ...   
s1mple          0.762649      0.564239       0.615256  ...   
w0nderful       0.093294      0.028255       0.051047  ...   
jL              0.196361      0.173509       0.169083  ...   
b1t             0.169153      0.141833       0.154725  ...   
iM              0.351496      0.338832       0.323825  ...   
Aleksib         0.357639      0.402564       0.422066  ...   
B1ad3           0.164812      0.258679       0.221556  ...   
sh1ro           0.413478      0.250494       0.356871  ...   
donk            0.076025      0.025594       0.014359  ...   
zont1x          0.077188      0.065101       0.064649  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
ZywOo                        0.425633                              0.505155   
Spinx                        0.011076                              0.013746   
mezii                        0.028481                              0.030928   
flameZ                       0.014241                              0.017182   
apEX                         0.020570                              0.020619   
s1mple                       0.455696                              0.532646   
w0nderful                    0.495253                              0.567010   
jL                           0.106013                              0.113402   
b1t                          0.012658                              0.013746   
iM                           0.011076                              0.013746   
Aleksib                      0.023734                              0.024055   
B1ad3                        0.012658                              0.010309   
sh1ro                        0.738924                              0.845361   
donk                         0.023734                              0.030928   
zont1x                       0.012658                              0.017182   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
ZywOo                         0.444444                          0.37500   
Spinx                         0.008889                        

age  rating       dpr       kpr      kast    impact       adr  \
player_name                                                                  
kinqie        23       0  0.761905  0.333333  0.361345  0.411765  0.417763   
boX           23       0  0.666667  0.300000  0.394958  0.514706  0.450658   
Prism         23       0  0.761905  0.333333  0.453782  0.397059  0.444079   
Noisia        23       0  0.714286  0.300000  0.378151  0.264706  0.371711   
arvid         23       0  0.761905  0.333333  0.352941  0.367647  0.371711   
xartE         23       0  0.571429  0.166667  0.369748  0.191176  0.269737   
kiR           23       0  0.809524  0.333333  0.210084  0.382353  0.345395   
Xp3           23       0  0.761905  0.266667  0.243697  0.323529  0.302632   
arki          23       0  0.666667  0.233333  0.394958  0.264706  0.302632   
sorrow        23       0  0.809524  0.300000  0.193277  0.264706  0.296053   
dobbo         23       0  0.523810  0.200000  0.462185  0.161765  0.184211   
S0tF1k        23       0  0.666667  0.233333  0.344538  0.294118  0.312500   
coolio        23       0  0.476190  0.100000  0.470588  0.058824  0.101974   
Psycho        23       0  0.714286  0.200000  0.361345  0.191176  0.282895   
bobz          23       0  0.761905  0.033333  0.268908  0.132353  0.151316   

             total_kills  total_deaths  rounds_played  ...  \
player_name                                            ...   
kinqie          0.148350      0.178298       0.155340  ...   
boX             0.445691      0.512468       0.504505  ...   
Prism           0.072206      0.089808       0.068749  ...   
Noisia          0.084826      0.101997       0.087049  ...   
arvid           0.251441      0.296868       0.278093  ...   
xartE           0.287378      0.347144       0.354795  ...   
kiR             0.195555      0.238863       0.210822  ...   
Xp3             0.041820      0.059272       0.040964  ...   
arki            0.188865      0.228574       0.221907  ...   
sorrow          0.036340      0.055927       0.034418  ...   
dobbo           0.051806      0.066900       0.064174  ...   
S0tF1k          0.398463      0.480462       0.471916  ...   
coolio          0.024314      0.039684       0.038923  ...   
Psycho          0.051142      0.078126       0.063452  ...   
bobz            0.006713      0.043231       0.024477  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
kinqie                       0.007911                              0.010309   
boX                          0.069620                              0.065292   
Prism                        0.007911                              0.010309   
Noisia                       0.017405                              0.017182   
arvid                        0.007911                              0.010309   
xartE                        0.139241                              0.134021   
kiR                          0.015823                              0.017182   
Xp3                          0.422468                              0.408935   
arki                         0.044304                              0.041237   
sorrow                       0.026899                              0.027491   
dobbo                        0.018987                              0.017182   
S0tF1k                       0.045886                              0.041237   
coolio                       0.004747                              0.003436   
Psycho                       0.009494                              0.006873   
bobz                         0.009494                              0.006873   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
kinqie                        0.006667                          0.00125   
boX                           0.066667                        

## Training the Data
- Determining X and y variables
- Utilizing train_test_split() to train and test data

In [108]:
training_team_players = team_players.copy()
X = training_team_players.drop(columns='rating')
display(X)

age       dpr       kpr      kast    impact       adr  \
player_name                                                          
ZywOo         23  0.333333  0.966667  0.924370  1.000000  0.855263   
Spinx         23  0.476190  0.600000  0.663866  0.529412  0.549342   
mezii         23  0.428571  0.500000  0.747899  0.441176  0.536184   
flameZ        23  0.571429  0.500000  0.689076  0.602941  0.549342   
apEX          23  0.714286  0.466667  0.352941  0.455882  0.523026   
...          ...       ...       ...       ...       ...       ...   
dobbo         23  0.523810  0.200000  0.462185  0.161765  0.184211   
S0tF1k        23  0.666667  0.233333  0.344538  0.294118  0.312500   
coolio        23  0.476190  0.100000  0.470588  0.058824  0.101974   
Psycho        23  0.714286  0.200000  0.361345  0.191176  0.282895   
bobz          23  0.761905  0.033333  0.268908  0.132353  0.151316   

             total_kills  total_deaths  rounds_played  kd_ratio  ...  \
player_name                                                      ...   
ZywOo           0.535451      0.360651       0.419567  0.924242  ...   
Spinx           0.181227      0.141909       0.152015  0.560606  ...   
mezii           0.208649      0.180832       0.196674  0.484848  ...   
flameZ          0.156083      0.141453       0.138413  0.424242  ...   
apEX            0.838105      0.891845       0.873289  0.333333  ...   
...                  ...           ...            ...       ...  ...   
dobbo           0.051806      0.066900       0.064174  0.227273  ...   
S0tF1k          0.398463      0.480462       0.471916  0.196970  ...   
coolio          0.024314      0.039684       0.038923  0.196970  ...   
Psycho          0.051142      0.078126       0.063452  0.151515  ...   
bobz            0.006713      0.043231       0.024477  0.045455  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
ZywOo                        0.425633                              0.505155   
Spinx                        0.011076                              0.013746   
mezii                        0.028481                              0.030928   
flameZ                       0.014241                              0.017182   
apEX                         0.020570                              0.020619   
...                               ...                                   ...   
dobbo                        0.018987                              0.017182   
S0tF1k                       0.045886                              0.041237   
coolio                       0.004747                              0.003436   
Psycho                       0.009494                              0.006873   
bobz                         0.009494                              0.006873   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
ZywOo                         0.444444                          0.37500   
Spinx                         0.008889                          0.00125   
mezii                         0.022222                          0.01250   
flameZ                        0.008889                          0.00250   
apEX                          0.011111                          0.02500   
...                                ...                              ...   
dobbo                         0.022222                          0.00750   
S0tF1k                        0.033333                          0.01250   
coolio                        0.005556                          0.00250   
Psycho                        0.008889                          0.00000   
bobz                          0.007778                          0.00250   

             utility_score  utility_damage_per_round  \
player_name                                            
ZywOo             0.609195                  0.180

In [109]:
y = team_players['rating']
display(y)

player_name
ZywOo     1
Spinx     1
mezii     1
flameZ    1
apEX      0
         ..
dobbo     0
S0tF1k    0
coolio    0
Psycho    0
bobz      0
Name: rating, Length: 585, dtype: int64

In [110]:
X_train, X_test, y_train, y_test = train_test_split(X, y)

In [111]:
display(X_train.head())
display(X_test.head())
display(y_train.head())
display(y_test.head())

age       dpr       kpr      kast    impact       adr  \
player_name                                                          
Keoz          23  0.761905  0.366667  0.310924  0.426471  0.398026   
yepz          23  0.523810  0.233333  0.563025  0.205882  0.250000   
Noktse        23  0.523810  0.366667  0.378151  0.455882  0.338816   
try           23  0.238095  0.666667  0.764706  0.632353  0.500000   
nython        23  0.428571  0.500000  0.588235  0.485294  0.338816   

             total_kills  total_deaths  rounds_played  kd_ratio  ...  \
player_name                                                      ...   
Keoz            0.316887      0.366048       0.341527  0.242424  ...   
yepz            0.110468      0.125944       0.126975  0.257576  ...   
Noktse          0.357473      0.373575       0.390463  0.348485  ...   
try             0.103091      0.046044       0.064508  0.742424  ...   
nython          0.246768      0.217222       0.236759  0.500000  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
Keoz                         0.041139                              0.044674   
yepz                         0.006329                              0.006873   
Noktse                       0.444620                              0.408935   
try                          0.606013                              0.635739   
nython                       0.536392                              0.525773   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
Keoz                          0.033333                          0.03750   
yepz                          0.007778                          0.00375   
Noktse                        0.333333                          0.37500   
try                           0.666667                          0.50000   
nython                        0.444444                          0.37500   

             utility_score  utility_damage_per_round  \
player_name                                            
Keoz              0.563218                  0.270358   
yepz              0.459770                  0.158523   
Noktse            0.804598                  0.263844   
try               0.701149                  0.166124   
nython            0.482759                  0.110749   

             utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
player_name                                                                   
Keoz                             0.506494                          0.491115   
yepz                             0.389610                          0.515347   
Noktse                           0.461039                          0.975767   
try                              0.493506                          0.862682   
nython                           0.266234                          0.652666   

             utility_flash_assists_per_round  \
player_name                                    
Keoz                                0.250000   
yepz                                0.333333   
Noktse                              0.916667   
try                                 0.916667   
nython                              0.666667   

             utility_time_opponent_flashed_per_round  
player_name                                           
Keoz                                        0.766784  
yepz                                        0.745583  
Noktse                                      0.823322  
try                                         0.855124  
nython                                      0.862191  

[5 rows x 64 columns]

age       dpr       kpr      kast    impact       adr  \
player_name                                                           
hades          23  0.380952  0.600000  0.571429  0.544118  0.404605   
Annihilation   23  0.333333  0.500000  0.621849  0.455882  0.394737   
mhL            23  0.428571  0.700000  0.596639  0.676471  0.625000   
Plopski        23  0.571429  0.466667  0.621849  0.441176  0.500000   
kreaz          23  0.523810  0.366667  0.554622  0.367647  0.453947   

              total_kills  total_deaths  rounds_played  kd_ratio  ...  \
player_name                                                       ...   
hades            0.350547      0.285642       0.324494  0.606061  ...   
Annihilation     0.055839      0.024327       0.033433  0.560606  ...   
mhL              0.254216      0.191551       0.208499  0.651515  ...   
Plopski          0.422160      0.421849       0.433134  0.393939  ...   
kreaz            0.253007      0.256399       0.267130  0.363636  ...   

              sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                    
hades                         0.694620                              0.728522   
Annihilation                  0.463608                              0.481100   
mhL                           0.699367                              0.769759   
Plopski                       0.009494                              0.010309   
kreaz                         0.047468                              0.051546   

              sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                                
hades                          0.666667                          0.62500   
Annihilation                   0.444444                          0.25000   
mhL                            0.777778                          0.62500   
Plopski                        0.007778                          0.00625   
kreaz                          0.033333                          0.02500   

              utility_score  utility_damage_per_round  \
player_name                                             
hades              0.494253                  0.107492   
Annihilation       0.586207                  0.135722   
mhL                0.528736                  0.232356   
Plopski            0.517241                  0.235613   
kreaz              0.712644                  0.326819   

              utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
player_name                                                                    
hades                             0.253247                          0.652666   
Annihilation                      0.305195                          0.749596   
mhL                               0.629870                          0.571890   
Plopski                           0.590909                          0.474960   
kreaz                             0.506494                          0.693053   

              utility_flash_assists_per_round  \
player_name                                     
hades                                0.583333   
Annihilation                         0.750000   
mhL                                  0.333333   
Plopski                              0.250000   
kreaz                                0.333333   

              utility_time_opponent_flashed_per_round  
player_name                                            
hades                                        0.883392  
Annihilation                                 0.865724  
mhL                                          0.763251  
Plopski                                      0.780919  
kreaz                                        0.763251  

[5 rows x 64 columns]

player_name
Keoz      0
yepz      0
Noktse    0
try       1
nython    1
Name: rating, dtype: int64

player_name
hades           1
Annihilation    1
mhL             1
Plopski         1
kreaz           0
Name: rating, dtype: int64

## Preparing the Model
- LogisticRegression

In [112]:
lr = LogisticRegression()

In [113]:
lr.fit(X_train, y_train)

LogisticRegression()

In [114]:
lr_teams_pred = lr.predict(X_test)
lr_teams_pred

array([1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0,
       1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1,
       1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0,
       0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1])

In [115]:
display(accuracy_score(y_test, lr_teams_pred))

0.9115646258503401

In [116]:
display(confusion_matrix(y_test, lr_teams_pred))

array([[64,  1],
       [12, 70]])

## LogisticRegression Model for No-Team
- Using solo-players dataframe

In [117]:
solo_players = pd.read_pickle('/Users/powellshayne/Desktop/githubrepos/VictorVis/CleanData/solo_players_df.pkl')
solo_players

real_name     team  age    rating       dpr       kpr  \
player_name                                                                 
SENSEi        Dmitriy Shvorak  no team   23  0.764706  0.047619  0.633333   
sterling           Euan Moore  no team   23  0.745098  0.285714  0.733333   
Jamppi         Elias Olkkonen  no team   23  0.666667  0.571429  0.733333   
Ethan            Ethan Arnold  no team   23  0.647059  0.619048  0.600000   
poizon       Valentin Vasilev  no team   23  0.647059  0.476190  0.600000   
...                       ...      ...  ...       ...       ...       ...   
seang@res          Sean Gares  no team   23  0.137255  0.666667  0.133333   
def           Dima Lemeshchuk  no team   23  0.098039  0.714286  0.133333   
advent             Zhuo Liang  no team   23  0.078431  0.476190  0.000000   
gob b             Fatih Dayik  no team   23  0.078431  0.619048  0.033333   
netrick       Jonas Wenzelsen  no team   23  0.019608  0.809524  0.033333   

                 kast    impact       adr  rating_is_missing  ...  \
player_name                                                   ...   
SENSEi       0.789916  0.617647  0.480263                  0  ...   
sterling     0.731092  0.632353  0.539474                  0  ...   
Jamppi       0.453782  0.779412  0.555921                  0  ...   
Ethan        0.647059  0.632353  0.661184                  0  ...   
poizon       0.512605  0.661765  0.542763                  0  ...   
...               ...       ...       ...                ...  ...   
seang@res    0.302521  0.176471  0.200658                  0  ...   
def          0.386555  0.147059  0.200658                  0  ...   
advent       0.537815  0.000000  0.000000                  0  ...   
gob b        0.319328  0.132353  0.171053                  0  ...   
netrick      0.159664  0.147059  0.180921                  0  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
SENSEi                       0.802215                              0.855670   
sterling                     0.580696                              0.632302   
Jamppi                       0.680380                              0.721649   
Ethan                        0.015823                              0.017182   
poizon                       0.594937                              0.601375   
...                               ...                                   ...   
seang@res                    0.052215                              0.089347   
def                          0.012658                              0.010309   
advent                       0.007911                              0.006873   
gob b                        0.037975                              0.024055   
netrick                      0.011076                              0.010309   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
SENSEi                        0.777778                          0.75000   
sterling                      0.555556                          0.50000   
Jamppi                        0.777778                          0.75000   
Ethan                         0.011111                          0.00500   
poizon                        0.555556                          0.50000   
...                                ...                              ...   
seang@res                     0.088889                          0.07500   
def                           0.010000                          0.00000   
advent                        0.003333                          0.00000   
gob b                         0.033333                          0.01250   
netrick                       0.008889                          0.00875   

             utility_score  utility_damage_per_round  \
player_name                                            
SENS

In [118]:
solo_players = solo_players.drop(columns = ['real_name', 'team', 'rating_is_missing'])

In [119]:
average_rating_solo = solo_players['rating'].mean()
average_rating_solo

0.4127886141401728

In [120]:
solo_players['rating'] = (solo_players['rating'] > average_rating_solo).astype(int)
solo_players

age  rating       dpr       kpr      kast    impact       adr  \
player_name                                                                  
SENSEi        23       1  0.047619  0.633333  0.789916  0.617647  0.480263   
sterling      23       1  0.285714  0.733333  0.731092  0.632353  0.539474   
Jamppi        23       1  0.571429  0.733333  0.453782  0.779412  0.555921   
Ethan         23       1  0.619048  0.600000  0.647059  0.632353  0.661184   
poizon        23       1  0.476190  0.600000  0.512605  0.661765  0.542763   
...          ...     ...       ...       ...       ...       ...       ...   
seang@res     23       0  0.666667  0.133333  0.302521  0.176471  0.200658   
def           23       0  0.714286  0.133333  0.386555  0.147059  0.200658   
advent        23       0  0.476190  0.000000  0.537815  0.000000  0.000000   
gob b         23       0  0.619048  0.033333  0.319328  0.132353  0.171053   
netrick       23       0  0.809524  0.033333  0.159664  0.147059  0.180921   

             total_kills  total_deaths  rounds_played  ...  \
player_name                                            ...   
SENSEi          0.137747      0.060919       0.103572  ...   
sterling        0.297578      0.202271       0.246419  ...   
Jamppi          0.088692      0.053596       0.046877  ...   
Ethan           0.291529      0.268258       0.267693  ...   
poizon          0.369595      0.317014       0.340630  ...   
...                  ...           ...            ...  ...   
seang@res       0.102972      0.145811       0.133838  ...   
def             0.063572      0.102808       0.088193  ...   
advent          0.290130      0.386752       0.414429  ...   
gob b           0.188510      0.267853       0.265018  ...   
netrick         0.004507      0.043333       0.020957  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
SENSEi                       0.802215                              0.855670   
sterling                     0.580696                              0.632302   
Jamppi                       0.680380                              0.721649   
Ethan                        0.015823                              0.017182   
poizon                       0.594937                              0.601375   
...                               ...                                   ...   
seang@res                    0.052215                              0.089347   
def                          0.012658                              0.010309   
advent                       0.007911                              0.006873   
gob b                        0.037975                              0.024055   
netrick                      0.011076                              0.010309   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
SENSEi                        0.777778                          0.75000   
sterling                      0.555556                          0.50000   
Jamppi                        0.777778                          0.75000   
Ethan                         0.011111                          0.00500   
poizon                        0.555556                          0.50000   
...                                ...                              ...   
seang@res                     0.088889                          0.07500   
def                           0.010000                          0.00000   
advent                        0.003333                          0.00000   
gob b                         0.033333                          0.01250   
netrick                       0.008889                          0.00875   

             utility_score  utility_damage_per_round  \
player_name                                            
SENSEi            0.632184                  0.174810   
sterling          0

In [121]:
solo_X= solo_players.copy().drop(columns='rating')

In [122]:
solo_y = solo_players['rating']

In [123]:
solo_X_train, solo_X_test, solo_y_train, solo_y_test = train_test_split(solo_X, solo_y)

In [124]:
sololr = LogisticRegression()

In [125]:
sololr.fit(solo_X_train, solo_y_train)

LogisticRegression()

In [126]:
solo_pred = sololr.predict(solo_X_test)
solo_pred

array([1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1,
       1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0,
       0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 1, 1, 1, 1])

In [127]:
display(accuracy_score(solo_y_test, solo_pred))

0.7916666666666666

In [128]:
display(confusion_matrix(solo_y_test, solo_pred))

array([[38,  8],
       [12, 38]])

## All Player Model
- Using dataframe with all players present

In [129]:
all_players

player_name         real_name      team  age    rating       dpr  \
0         ZywOo   Mathieu Herbaut  vitality   23  1.000000  0.333333   
1         Spinx      Lotan Giladi  vitality   23  0.627451  0.476190   
2         mezii  William Merriman  vitality   23  0.588235  0.428571   
3        flameZ    Shahar Shushan  vitality   23  0.607843  0.571429   
4          apEX   Dan Madesclaire  vitality   23  0.372549  0.714286   
..          ...               ...       ...  ...       ...       ...   
963   seang@res        Sean Gares   no team   23  0.137255  0.666667   
964         def   Dima Lemeshchuk   no team   23  0.098039  0.714286   
965      advent        Zhuo Liang   no team   23  0.078431  0.476190   
966       gob b       Fatih Dayik   no team   23  0.078431  0.619048   
967     netrick   Jonas Wenzelsen   no team   23  0.019608  0.809524   

          kpr      kast    impact       adr  ...  sniping_kills_percentage  \
0    0.966667  0.924370  1.000000  0.855263  ...                  0.425633   
1    0.600000  0.663866  0.529412  0.549342  ...                  0.011076   
2    0.500000  0.747899  0.441176  0.536184  ...                  0.028481   
3    0.500000  0.689076  0.602941  0.549342  ...                  0.014241   
4    0.466667  0.352941  0.455882  0.523026  ...                  0.020570   
..        ...       ...       ...       ...  ...                       ...   
963  0.133333  0.302521  0.176471  0.200658  ...                  0.052215   
964  0.133333  0.386555  0.147059  0.200658  ...                  0.012658   
965  0.000000  0.537815  0.000000  0.000000  ...                  0.007911   
966  0.033333  0.319328  0.132353  0.171053  ...                  0.037975   
967  0.033333  0.159664  0.147059  0.180921  ...                  0.011076   

     sniping_rounds_with_kills_percentage  sniping_multi_kill_rounds  \
0                                0.505155                   0.444444   
1                                0.013746                   0.008889   
2                                0.030928                   0.022222   
3                                0.017182                   0.008889   
4                                0.020619                   0.011111   
..                                    ...                        ...   
963                              0.089347                   0.088889   
964                              0.010309                   0.010000   
965                              0.006873                   0.003333   
966                              0.024055                   0.033333   
967                              0.010309                   0.008889   

     sniping_opening_kills_per_round  utility_score  utility_damage_per_round  \
0                            0.37500       0.609195                  0.180239   
1                            0.00125       0.252874                  0.173724   
2                            0.01250       0.793103                  0.277959   
3                            0.00250       0.839080                  0.302932   
4                            0.02500       0.551724                  0.232356   
..                               ...            ...                       ...   
963                          0.07500       0.333333                  0.292074   
964                          0.00000       0.563218                  0.146580   
965                          0.00000       0.735632                  0.205212   
966                          0.01250       0.747126                  0.280130   
967                          0.00875       0.873563                  0.422367   

     utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
0                        0.474026                          0.652666   
1                        0.428571                          0.410339   
2                        0.422078                          0.781906   
3                        0.500000                          0.765751   
4        

In [130]:
all_players = all_players.drop(columns= ['real_name', 'team', 'rating_is_missing'])
all_players

player_name  age    rating       dpr       kpr      kast    impact  \
0         ZywOo   23  1.000000  0.333333  0.966667  0.924370  1.000000   
1         Spinx   23  0.627451  0.476190  0.600000  0.663866  0.529412   
2         mezii   23  0.588235  0.428571  0.500000  0.747899  0.441176   
3        flameZ   23  0.607843  0.571429  0.500000  0.689076  0.602941   
4          apEX   23  0.372549  0.714286  0.466667  0.352941  0.455882   
..          ...  ...       ...       ...       ...       ...       ...   
963   seang@res   23  0.137255  0.666667  0.133333  0.302521  0.176471   
964         def   23  0.098039  0.714286  0.133333  0.386555  0.147059   
965      advent   23  0.078431  0.476190  0.000000  0.537815  0.000000   
966       gob b   23  0.078431  0.619048  0.033333  0.319328  0.132353   
967     netrick   23  0.019608  0.809524  0.033333  0.159664  0.147059   

          adr  total_kills  total_deaths  ...  sniping_kills_percentage  \
0    0.855263     0.535451      0.360651  ...                  0.425633   
1    0.549342     0.181227      0.141909  ...                  0.011076   
2    0.536184     0.208649      0.180832  ...                  0.028481   
3    0.549342     0.156083      0.141453  ...                  0.014241   
4    0.523026     0.838105      0.891845  ...                  0.020570   
..        ...          ...           ...  ...                       ...   
963  0.200658     0.102972      0.145811  ...                  0.052215   
964  0.200658     0.063572      0.102808  ...                  0.012658   
965  0.000000     0.290130      0.386752  ...                  0.007911   
966  0.171053     0.188510      0.267853  ...                  0.037975   
967  0.180921     0.004507      0.043333  ...                  0.011076   

     sniping_rounds_with_kills_percentage  sniping_multi_kill_rounds  \
0                                0.505155                   0.444444   
1                                0.013746                   0.008889   
2                                0.030928                   0.022222   
3                                0.017182                   0.008889   
4                                0.020619                   0.011111   
..                                    ...                        ...   
963                              0.089347                   0.088889   
964                              0.010309                   0.010000   
965                              0.006873                   0.003333   
966                              0.024055                   0.033333   
967                              0.010309                   0.008889   

     sniping_opening_kills_per_round  utility_score  utility_damage_per_round  \
0                            0.37500       0.609195                  0.180239   
1                            0.00125       0.252874                  0.173724   
2                            0.01250       0.793103                  0.277959   
3                            0.00250       0.839080                  0.302932   
4                            0.02500       0.551724                  0.232356   
..                               ...            ...                       ...   
963                          0.07500       0.333333                  0.292074   
964                          0.00000       0.563218                  0.146580   
965                          0.00000       0.735632                  0.205212   
966                          0.01250       0.747126                  0.280130   
967                          0.00875       0.873563                  0.422367   

     utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
0                        0.474026                          0.652666   
1                        0.428571                          0.410339   
2                        0.422078                          0.781906   
3                        0.500000                          0.765751   
4                    

In [131]:
average_rating_all = all_players['rating'].mean()
average_rating_all

0.45440366229136264

In [132]:
all_players['rating'] = (all_players['rating'] > average_rating_all).astype(int)
all_players

player_name  age  rating       dpr       kpr      kast    impact  \
0         ZywOo   23       1  0.333333  0.966667  0.924370  1.000000   
1         Spinx   23       1  0.476190  0.600000  0.663866  0.529412   
2         mezii   23       1  0.428571  0.500000  0.747899  0.441176   
3        flameZ   23       1  0.571429  0.500000  0.689076  0.602941   
4          apEX   23       0  0.714286  0.466667  0.352941  0.455882   
..          ...  ...     ...       ...       ...       ...       ...   
963   seang@res   23       0  0.666667  0.133333  0.302521  0.176471   
964         def   23       0  0.714286  0.133333  0.386555  0.147059   
965      advent   23       0  0.476190  0.000000  0.537815  0.000000   
966       gob b   23       0  0.619048  0.033333  0.319328  0.132353   
967     netrick   23       0  0.809524  0.033333  0.159664  0.147059   

          adr  total_kills  total_deaths  ...  sniping_kills_percentage  \
0    0.855263     0.535451      0.360651  ...                  0.425633   
1    0.549342     0.181227      0.141909  ...                  0.011076   
2    0.536184     0.208649      0.180832  ...                  0.028481   
3    0.549342     0.156083      0.141453  ...                  0.014241   
4    0.523026     0.838105      0.891845  ...                  0.020570   
..        ...          ...           ...  ...                       ...   
963  0.200658     0.102972      0.145811  ...                  0.052215   
964  0.200658     0.063572      0.102808  ...                  0.012658   
965  0.000000     0.290130      0.386752  ...                  0.007911   
966  0.171053     0.188510      0.267853  ...                  0.037975   
967  0.180921     0.004507      0.043333  ...                  0.011076   

     sniping_rounds_with_kills_percentage  sniping_multi_kill_rounds  \
0                                0.505155                   0.444444   
1                                0.013746                   0.008889   
2                                0.030928                   0.022222   
3                                0.017182                   0.008889   
4                                0.020619                   0.011111   
..                                    ...                        ...   
963                              0.089347                   0.088889   
964                              0.010309                   0.010000   
965                              0.006873                   0.003333   
966                              0.024055                   0.033333   
967                              0.010309                   0.008889   

     sniping_opening_kills_per_round  utility_score  utility_damage_per_round  \
0                            0.37500       0.609195                  0.180239   
1                            0.00125       0.252874                  0.173724   
2                            0.01250       0.793103                  0.277959   
3                            0.00250       0.839080                  0.302932   
4                            0.02500       0.551724                  0.232356   
..                               ...            ...                       ...   
963                          0.07500       0.333333                  0.292074   
964                          0.00000       0.563218                  0.146580   
965                          0.00000       0.735632                  0.205212   
966                          0.01250       0.747126                  0.280130   
967                          0.00875       0.873563                  0.422367   

     utility_kills_per_100_rounds  utility_flashes_thrown_per_round  \
0                        0.474026                          0.652666   
1                        0.428571                          0.410339   
2                        0.422078                          0.781906   
3                        0.500000                          0.765751   
4                        0.311688            

In [133]:
all_players = all_players.set_index('player_name')

In [134]:
all_X = all_players.copy().drop(columns= 'rating')
all_X

age       dpr       kpr      kast    impact       adr  \
player_name                                                          
ZywOo         23  0.333333  0.966667  0.924370  1.000000  0.855263   
Spinx         23  0.476190  0.600000  0.663866  0.529412  0.549342   
mezii         23  0.428571  0.500000  0.747899  0.441176  0.536184   
flameZ        23  0.571429  0.500000  0.689076  0.602941  0.549342   
apEX          23  0.714286  0.466667  0.352941  0.455882  0.523026   
...          ...       ...       ...       ...       ...       ...   
seang@res     23  0.666667  0.133333  0.302521  0.176471  0.200658   
def           23  0.714286  0.133333  0.386555  0.147059  0.200658   
advent        23  0.476190  0.000000  0.537815  0.000000  0.000000   
gob b         23  0.619048  0.033333  0.319328  0.132353  0.171053   
netrick       23  0.809524  0.033333  0.159664  0.147059  0.180921   

             total_kills  total_deaths  rounds_played  kd_ratio  ...  \
player_name                                                      ...   
ZywOo           0.535451      0.360651       0.419567  0.924242  ...   
Spinx           0.181227      0.141909       0.152015  0.560606  ...   
mezii           0.208649      0.180832       0.196674  0.484848  ...   
flameZ          0.156083      0.141453       0.138413  0.424242  ...   
apEX            0.838105      0.891845       0.873289  0.333333  ...   
...                  ...           ...            ...       ...  ...   
seang@res       0.102972      0.145811       0.133838  0.136364  ...   
def             0.063572      0.102808       0.088193  0.106061  ...   
advent          0.290130      0.386752       0.414429  0.106061  ...   
gob b           0.188510      0.267853       0.265018  0.090909  ...   
netrick         0.004507      0.043333       0.020957  0.030303  ...   

             sniping_kills_percentage  sniping_rounds_with_kills_percentage  \
player_name                                                                   
ZywOo                        0.425633                              0.505155   
Spinx                        0.011076                              0.013746   
mezii                        0.028481                              0.030928   
flameZ                       0.014241                              0.017182   
apEX                         0.020570                              0.020619   
...                               ...                                   ...   
seang@res                    0.052215                              0.089347   
def                          0.012658                              0.010309   
advent                       0.007911                              0.006873   
gob b                        0.037975                              0.024055   
netrick                      0.011076                              0.010309   

             sniping_multi_kill_rounds  sniping_opening_kills_per_round  \
player_name                                                               
ZywOo                         0.444444                          0.37500   
Spinx                         0.008889                          0.00125   
mezii                         0.022222                          0.01250   
flameZ                        0.008889                          0.00250   
apEX                          0.011111                          0.02500   
...                                ...                              ...   
seang@res                     0.088889                          0.07500   
def                           0.010000                          0.00000   
advent                        0.003333                          0.00000   
gob b                         0.033333                          0.01250   
netrick                       0.008889                          0.00875   

             utility_score  utility_damage_per_round  \
player_name                                            
ZywOo             0.609195                  0.180

In [135]:
all_y = all_players['rating']
all_y

player_name
ZywOo        1
Spinx        1
mezii        1
flameZ       1
apEX         0
            ..
seang@res    0
def          0
advent       0
gob b        0
netrick      0
Name: rating, Length: 968, dtype: int64

In [136]:
all_X_train, all_X_test, all_y_train, all_y_test = train_test_split(all_X, all_y)

In [137]:
all_lr = LogisticRegression()

In [138]:
all_lr.fit(all_X_train, all_y_train)

LogisticRegression()

In [139]:
all_player_pred = all_lr.predict(all_X_test)
all_player_pred

array([1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0,
       0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1,
       1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0,
       0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0,
       1, 0, 1, 0, 1, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0,
       0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 1, 1])

In [140]:
display(accuracy_score(all_y_test, all_player_pred))

0.9049586776859504

In [141]:
display(confusion_matrix(all_y_test, all_player_pred))

array([[125,   8],
       [ 15,  94]])

## Full Data Results

- Team Players -- 0.9115646258503401 Accuracy Score
- Solo Players -- 0.7916666666666666 Accuracy Score
- All Players -- 0.9049586776859504 Accuracy Score


 *The accuracy scores for the logistic regression model for each collection of data; team, solo, and all; has resulted with the Team Players having a higher accuracy score than even the collection of all players*

## Logistic Regression (Corr. Features Only)
- Features utilized here are determined in ExploreData.ipynb
- kd_ratio
- kpr
- firepower_score
- firepower_rounds_with_kill
- impact

In [171]:
team_players_featured = team_players[['rating', 'kd_ratio', 'kpr', 'firepower_score', 'firepower_rounds_with_kill', 'impact']]
team_players_featured

rating  kd_ratio       kpr  firepower_score  \
player_name                                                
ZywOo             1  0.924242  0.966667         1.000000   
Spinx             1  0.560606  0.600000         0.788889   
mezii             1  0.484848  0.500000         0.677778   
flameZ            1  0.424242  0.500000         0.711111   
apEX              0  0.333333  0.466667         0.666667   
...             ...       ...       ...              ...   
dobbo             0  0.227273  0.200000         0.177778   
S0tF1k            0  0.196970  0.233333         0.311111   
coolio            0  0.196970  0.100000         0.100000   
Psycho            0  0.151515  0.200000         0.222222   
bobz              0  0.045455  0.033333         0.077778   

             firepower_rounds_with_kill    impact  
player_name                                        
ZywOo                          0.881579  1.000000  
Spinx                          0.631579  0.529412  
mezii                          0.539474  0.441176  
flameZ                         0.559211  0.602941  
apEX                           0.473684  0.455882  
...                                 ...       ...  
dobbo                          0.276316  0.161765  
S0tF1k                         0.282895  0.294118  
coolio                         0.223684  0.058824  
Psycho                         0.269737  0.191176  
bobz                           0.072368  0.132353  

[585 rows x 6 columns]

In [172]:
solo_players_featured = solo_players[['rating', 'kd_ratio', 'kpr', 'firepower_score', 'firepower_rounds_with_kill', 'impact']]
solo_players_featured

rating  kd_ratio       kpr  firepower_score  \
player_name                                                
SENSEi            1  0.863636  0.633333         0.755556   
sterling          1  0.772727  0.733333         0.844444   
Jamppi            1  0.575758  0.733333         0.855556   
Ethan             1  0.469697  0.600000         0.811111   
poizon            1  0.560606  0.600000         0.788889   
...             ...       ...       ...              ...   
seang@res         0  0.136364  0.133333         0.166667   
def               0  0.106061  0.133333         0.144444   
advent            0  0.106061  0.000000         0.000000   
gob b             0  0.090909  0.033333         0.077778   
netrick           0  0.030303  0.033333         0.088889   

             firepower_rounds_with_kill    impact  
player_name                                        
SENSEi                         0.717105  0.617647  
sterling                       0.769737  0.632353  
Jamppi                         0.684211  0.779412  
Ethan                          0.677632  0.632353  
poizon                         0.539474  0.661765  
...                                 ...       ...  
seang@res                      0.144737  0.176471  
def                            0.177632  0.147059  
advent                         0.085526  0.000000  
gob b                          0.092105  0.132353  
netrick                        0.125000  0.147059  

[383 rows x 6 columns]

In [174]:
tf_X = team_players_featured.drop(columns='rating')
tf_y = team_players_featured['rating']

sf_X = solo_players_featured.drop(columns='rating')
sf_y = solo_players_featured['rating']

In [175]:
tf_X_train, tf_X_test, tf_y_train, tf_y_test = train_test_split(tf_X, tf_y)

sf_X_train, sf_X_test, sf_y_train, sf_y_test = train_test_split(sf_X, sf_y)

In [176]:
tf = LogisticRegression()
sf = LogisticRegression()

tf.fit(tf_X_train, tf_y_train)
sf.fit(sf_X_train, sf_y_train)

tf_predictions = tf.predict(tf_X_test)
sf_predictions = sf.predict(sf_X_test)

print(f'Team Feature Accuracy Score: {accuracy_score(tf_y_test, tf_predictions)}')
print(f'Solo Feature Accuracy Score: {accuracy_score(sf_y_test, sf_predictions)}')

Team Feature Accuracy Score: 0.8775510204081632
Solo Feature Accuracy Score: 0.7708333333333334
